src/main/java/core/basesyntax/reportgeneratorservice/RecordBalance.java

```diff
@@ -0,0 +1,39 @@
+package core.basesyntax.reportgeneratorservice;
+
+import core.basesyntax.filevalidatorservice.LineSpliterator;
+import core.basesyntax.filevalidatorservice.Validator;
+import core.basesyntax.storeactivities.ActionFunction;
+import core.basesyntax.storeactivities.StoreFunction;
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.HashMap;
+
+public class RecordBalance implements Validator, ReportGenerator {
+    private LineSpliterator lineSpliterator = new LineSpliterator();
+    private ActionFunction actionFunction = new ActionFunction();
+
+    public void recordBalance(String sourceFile, String reportFile) {
+        HashMap<String, Integer> balance = new HashMap<>();
+        try (BufferedReader bufferedReader = new BufferedReader(new FileReader(sourceFile));
```

Comment:
why do you read the file twice in a validator and here? If you create a FileReaderService you should read the file only once and parse your data further into other methods